script.js

```diff
@@ -1,12 +1,210 @@
+    )} - ${episode.name}`;
+
+    selectMenu.appendChild(optionEl);
+  });
+  drawEpiCard();
+}
+function refreshShowMenu() {
+  const selectShow = document.getElementById(_showDDLCtrlId);
+  selectShow.innerHTML = "";
+  //add options to ddl(select show)
+  _localShows.forEach((show) => {
+    const showEl = document.createElement("option");
+    showEl.innerText = show.name;
+    showEl.value = show.id;
+    selectShow.appendChild(showEl);
+  });
+}
+function createSearchResult() {
+  const rootElem = document.getElementById("root");
+  const resultP = document.createElement("p");
+  resultP.innerHTML = `Got ${_localEpisodes.length} episode(s). The data has (originally) come from <a href= "https://www.tvmaze.com/">TVMaze.com</a><br>`;
+  rootElem.appendChild(resultP);
+}
+function createShowMenu() {
+  const rootElem = document.getElementById("root");
+
+  // select/ddl (select show)
+  const selectShow = document.createElement("select");
+  selectShow.id = _showDDLCtrlId;
+  selectShow.addEventListener("change", function () {
+    refreshEpisodesDDL(selectShow.value);
+  });
+
+  rootElem.appendChild(selectShow);
 }
+function createEPMenu() {
+  const rootElem = document.getElementById("root");
+  const selectMenu = document.createElement("select");
+  selectMenu.id = _epDDLCtrlId;
+  selectMenu.addEventListener("change", function () {
+    drawEpiCard();
+  });
+  rootElem.appendChild(selectMenu);
+}
+function createSearchBar() {
+  const rootElem = document.getElementById("root");
+
+  //search bar
+  const searchBar = document.createElement("input");
+  searchBar.id = "txtSearchBy";
+  searchBar.setAttribute("type", "string");
+  searchBar.addEventListener("input", function () {
+    drawEpiCard();
+  });
+
+  //search result
+  const searchResult = document.createElement("label");
+  searchResult.id = "searchResult";
+
+  // const resultP = document.createElement("p");
+  // resultP.innerHTML = `Got ${_localEpisodes.length} episode(s). The data has (originally) come from <a href= "https://www.tvmaze.com/">TVMaze.com</a><br>`;
+  // //rootElem.innerHTML = `Got ${_localEpisodes.length} episode(s). The data has (originally) come from <a href= "https://www.tvmaze.com/">TVMaze.com</a><br>`;
+
+  // rootElem.insertBefore(resultP, rootElem.firstChild);
+  rootElem.appendChild(searchBar);
+  rootElem.appendChild(searchResult);
+  //drawEpiCard();
+}
+function clearCards() {
+  document
+    .querySelectorAll("#root div.grid-container")
+    .forEach((el) => el.remove());
+}
+function drawEpiCard() {
+  clearCards();
+  //get search words
+  const searchBar = document.getElementById("txtSearchBy");
+  let searchBy = searchBar.value.toLowerCase();
 
-function makePageForEpisodes(episodeList) {
+  //root
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+
+  //selected option
+  const selectMenu = document.getElementById("selName");
+  let selectedName = selectMenu.value;
```

Comment:
here it would be good to be more specific with your variable names, as you're using 2 different `select` menus in your app, for example `const selectEpisodeMenu`